src/modules/NewsList/index.tsx

```diff
@@ -72,34 +101,33 @@ export function NewsListUI(props: NewsListUIProps): ReactElement {
-                                                        <div className="rounded-full bg-gray-200 h-[10px] mr-2 w-16 mt-2" />
-                                                        <div className="rounded-full bg-gray-200 h-[10px] mr-2 w-10 mt-2" />
-                                                    </div>
-                                                </div>
-                                            </div>
-                                        </li>
-                                    ))}
-                                </ul>
+                                <ul className="w-full NewsList__loading">{loader(15)}</ul>
                             ))
                             .with({ status: 'paused' }, () => wrapMsg('There is no news.'))
                             .with({ status: 'error' }, () => wrapMsg('There was an error loading news.'))
                             .with({ status: 'empty' }, () => wrapMsg('There is no news.'))
-                            .with({ status: 'success' }, ({ data }) => (
-                                <ul className="pt-1.5 w-full">
+                            .with({ status: 'success' }, { status: 'refetching' }, ({ data, status }) => (
+                                <ul className="py-1.5 w-full">
+                                    <div
+                                        className={classNames(
+                                            'animate-pulse cursor-pointer duration-200 ease-in-out flex group items-center justify-center transition-h hover:animate-none',
+                                            {
+                                                invisible: !canRefetch,
+                                                'h-0': !canRefetch,
+                                                'pb-3': canRefetch,
+                                                'pt-2': canRefetch,
+                                            }
+                                        )}
+                                        onClick={onRefetch}
+                                    >
+                                        <div className="ml-2 w-full flex h-[1px] bg-gradient-to-l from-gray-200 group-hover:from-gray-300" />
+                                        <p className="px-3 text-gray-500 text-sm whitespace-nowrap group-hover:text-gray-700">
+                                            Check for new articles
+                                        </p>
+                                        <div className="mr-2 w-full flex h-[1px] bg-gradient-to-r from-gray-200 group-hover:from-gray-300" />
+                                    </div>
                                     {data.search.content.hits.map((hit) => {
-                                        const { content, id: newsId } = hit;
+                                        const content = hit.content as NewsContent;
```

Comment:
Do we want a cast here? Casting it to that type will make it look like all the fields are available vs. only what was queried.

Reply:
I wasn't sure how to tell TS that the search hits are `NewsContent`, and not `FilingContent`, for example. That's the only reason for the cast. Open to suggestions...

Reply:
So still not perfect, but the `NewListNews` type from above might cover this (though it should already be that type). What's the error if you don't do this?

Another possible solution is to write a function like `getContent` that takes in a NewsListQuery hit, checks the typename field for `NewsContent` and returns it (or undefined/null otherwise). This function should then have a type that is just the queried news or undefined. https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards

Reply:
The error isn't on `content` itself, but `newsSource` below (line 176 in this file).

```
TS2339: Property 'newsSource' does not exist on type '{ __typename?: "FilingContent...
```

because `newsSource` is only defined for `NewsContent`, so in the query...

```
                                ... on NewsContent {
                                    newsSource {
                                        id
                                        name
                                    }
                                }
```

So I think your suggestion of using a type guard function should take care of this.